### PR TITLE
[#4227] improvement(client-python): Add Catalog Error Handler and related exceptions, test cases in client-python

### PR DESCRIPTION
**Pull request by jingjia88**
### What changes were proposed in this pull request?

- Add Catalog Error Handler and related exceptions, test cases in `client-python` based on `client-java`
- Add lacking ITs in `test_catalog.py`

### Why are the changes needed?

Fix: #4227 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT Added and test with `./gradlew client:client-python:test`

### CODE REVIEW

**Comment by xunliu** [comment]
hi @noidname01 Please help review this PR, thanks!

**Comment by jerryshao** [comment]
@jingjia88 would you please fix the conflicts?

File: clients/client-python/tests/unittests/test_error_handler.py [hunk 1]
```diff
@@ -151,6 +155,46 @@ def test_metalake_error_handler(self):
...
+            CATALOG_ERROR_HANDLER.handle(
+                ErrorResponse.generate_error_response(
+                    NoSuchCatalogException, "mock error"
+                )
+            )
+
+        with self.assertRaises(CatalogAlreadyExistsException):
+            CATALOG_ERROR_HANDLER.handle(
+                ErrorResponse.generate_error_response(
+                    CatalogAlreadyExistsException, "mock error"
```
Comment (noidname01) [inline clients/client-python/tests/unittests/test_error_handler.py:185]: There are two right parenthesis missing.

File: clients/client-python/gravitino/exceptions/base.py [hunk 2]
```diff
@@ -95,6 +95,18 @@ class UnknownError(RuntimeError):
     """An exception thrown when other unknown exception is thrown"""
 
 
+class ConnectionFailedException(GravitinoRuntimeException):
+    """An exception thrown when connect to catalog failed."""
+
+
+class NoSuchCatalogException(NotFoundException):
```
Comment (noidname01) [inline clients/client-python/gravitino/exceptions/base.py:102]: This class is redefined.

**Review by noidname01 — CHANGES_REQUESTED** [review]
It seems that there are some problems when you solve the merge conflict, please remember to run test locally before every push.

**Comment by jingjia88** [comment]
Finished~

**Review by noidname01 — CHANGES_REQUESTED** [review]
Except for the minor issue, LGTM.

File: clients/client-python/gravitino/exceptions/base.py [hunk 3]
```diff
@@ -95,10 +95,17 @@ class UnknownError(RuntimeError):
     """An exception thrown when other unknown exception is thrown"""
 
 
+class ConnectionFailedException(GravitinoRuntimeException):
+    """An exception thrown when connect to catalog failed."""
+
+
+class CatalogAlreadyExistsException(AlreadyExistsException):
```
[the remark this answers is not in the document]
Comment (jingjia88) [reply clients/client-python/gravitino/exceptions/base.py]: Done!

**Review by noidname01 — APPROVED** [review]

**Review by jerryshao — APPROVED** [review]

**Review by mchades — APPROVED** [review]
I'm going to merge this PR, thanks for your contributions!

**Pull request merged**